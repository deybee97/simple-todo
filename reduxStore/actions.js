
export const UPDATE_TODO ='UPDATE_TODO'

export const UPDATE_USER =  'UPDATE_USER'


export const todoAction = (update) => ({
    type: UPDATE_TODO,
    payload: update
})

export const userAction = (update) => ({
    type: UPDATE_USER,
    payload: update
})






