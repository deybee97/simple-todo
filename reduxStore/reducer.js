import { combineReducers} from 'redux'

import { UPDATE_TODO, UPDATE_USER, LOG_IN_FULFILLED, LOG_IN_REJECTED, LOG_IN_SENT} from './actions'


const todoReducer = (state = [], action) => {

    switch(action.type){
    
        case UPDATE_TODO: return [ ...state, action.payload]
        default: return state
       
    }

}

const userReducer = (state={}, action) => {

    switch(action.type){
        case UPDATE_USER: return { ...state, ...action.payload}
        case LOG_IN_FULFILLED: return {...state, ...action.payload}
        default: return state
    }
}


export default combineReducers({
    todos: todoReducer,
    user: userReducer
})

