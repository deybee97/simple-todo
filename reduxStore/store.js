import { configureStore } from '@reduxjs/toolkit';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import reducer from './reducer';
import thunk from 'redux-thunk'; // Import redux-thunk

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: [thunk], // Use redux-thunk as middleware
});

export const persistor = persistStore(store);

// You can still dispatch actions as before
// store.dispatch(userAction({ username: 'Adebola', previlege: 'Admin' }));
// store.dispatch(todoAction('do this'));
// store.dispatch(todoAction('do that'));

// export default store;
