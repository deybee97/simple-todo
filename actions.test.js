import * as actions from "./reduxStore/actions"




describe('should return an action',()=>{
    it('should return an action', () => {
        expect(actions.userAction({userId:"12345609"})).toMatchSnapshot()
      })
    
      it('should handle empty object',()=>{
        expect(actions.userAction({})).toMatchSnapshot()
      })

      it('should handle empty name',()=>{
        expect(actions.userAction({name:""})).toMatchSnapshot()
      })
})

test('should return an action', () => {
  expect(actions.todoAction("do this")).toMatchSnapshot()
})

describe('login user returns action', ()=>{

    const mocLogin = async (email,password)=> {
            
        if(email === "username" && password === "password"){
            return "thisisatesttoken"
        }else{
            throw new Error("credentials incorrect")
        }
    }

    it('dispatches LOG_IN_SENT',    async()=>{

        const mockDispatch = jest.fn()

        await actions.loginUser("","")(mockDispatch)
       console.log(mockDispatch.mock.calls)
        expect(mockDispatch.mock.calls[0][0]).toEqual({type: actions.LOG_IN_SENT})
    } )


   

    it('dispatches LOG_IN_FULFILLED',async()=>{

        const mockDispatch = jest.fn()
       

        await actions.loginUser("username","password",mocLogin)(mockDispatch)

        console.log(mockDispatch.mock.calls)

        expect(mockDispatch.mock.calls[1][0]).toEqual({type: actions.LOG_IN_FULFILLED, payload:{token:"thisisatesttoken"}})
        expect(mockDispatch.mock.calls[1]).toMatchSnapshot()
    })

    it('dispatches LOG_IN_REJECTED',async()=>{

        const mockDispatch = jest.fn()
      

        await actions.loginUser("","",mocLogin)(mockDispatch)

        console.log(mockDispatch.mock.calls)

        expect(mockDispatch.mock.calls[1][0]).toEqual({type: actions.LOG_IN_REJECTED, payload:{errMsg:"credentials incorrect"}})
        expect(mockDispatch.mock.calls[1]).toMatchSnapshot()
    })


    
    
})



