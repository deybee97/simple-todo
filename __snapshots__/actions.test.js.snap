// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login user returns action dispatches LOG_IN_FULFILLED 1`] = `
[
  {
    "payload": {
      "token": "thisisatesttoken",
    },
    "type": "LOG_IN_FULFILLED",
  },
]
`;

exports[`should return an action 1`] = `
{
  "payload": "do this",
  "type": "UPDATE_TODO",
}
`;

exports[`should return an action should handle empty name 1`] = `
{
  "payload": {
    "name": "",
  },
  "type": "UPDATE_USER",
}
`;

exports[`should return an action should handle empty object 1`] = `
{
  "payload": {},
  "type": "UPDATE_USER",
}
`;

exports[`should return an action should hangle empty object 1`] = `
{
  "payload": {},
  "type": "UPDATE_USER",
}
`;

exports[`should return an action should return an action 1`] = `
{
  "payload": {
    "userId": "12345609",
  },
  "type": "UPDATE_USER",
}
`;
